src/contracts/libraries/GPv2Encoding.sol

```diff
@@ -131,6 +130,7 @@ library GPv2Encoding {
         bytes32 r = abi.decode(encodedOrder[140:], (bytes32));
         bytes32 s = abi.decode(encodedOrder[172:], (bytes32));
 
+        order.typeHash = ORDER_TYPE_HASH;
```

Comment:
Not sure if this get optimised away but this wouldn't even have to be set more than once in total (thus I'd expect this to actually be marginally cheaper as amount of orders grow).

Reply:
I am not sure I understand why `typeHash` is being set to a constant value here. What is the purpose of having a field in each order with the same data for the first few bytes?

Reply:
This field is used as part of the EIP-712 structured data signing. Specifically, its a hash that is unique per type definition and that needs to prefix all of the data that needs to be signed.